backend/file/file.go

```diff
@@ -14,21 +15,24 @@ import (
 // Backend that loads a configuration from a file.
 // It supports json and yaml formats.
 type Backend struct {
-	path string
-	name string
+	path     string
+	name     string
+	optional bool
 }
 
 // NewBackend creates a configuration loader that loads from a file.
 // The content will get decoded based on the file extension.
-func NewBackend(path string) *Backend {
+// If optional parameter is set to true, calling Unmarshal won't return an error if the file doesn't exist.
+func NewBackend(path string, optional bool) *Backend {
```

Comment:
Would it make sense to define an `.Optional(bool) *Backend` method instead in order to avoid breaking changes and keep it easily extendable?

The backend would then be called with `file.NewBackend("myfile.yaml").Optional(true)`.

Reply:
As the major version is still 0, the API of Confita is not considered as stable thus any breaking changes can happen. (cf https://semver.org/#spec-item-4).

Reply:
In general I'm [not that keen on boolean arguments](https://martinfowler.com/bliki/FlagArgument.html).

Also, I wonder what the good default is here. Perhaps we'd like it to be robust in the face of a missing file in the usual case.

Some possibilities:

    // IgnoreMissingFile configures the backend so that if the file is
    // missing, it will behave as if the data is not found instead of failing.
    func (b *Backend) IgnoreMissingFile()

    // NewOptionalBackend returns a backend that does not fail
    // if the file with the given path does not exist.
    func NewOptionalBackend(path string) *Backend

Also, even though the major version is still 0, from bitter experience it's still better to be compatible unless there's a compelling reason not to be.

Reply:
Thank you for the sharing @rogpeppe it is a nice article.
I opt with the 2nd possibility (`NewOptionalBackend`) for the two following reasons:
- it still allows to chain method call: 
```go
confita.NewLoader(file. NewOptionalBackend(...), env.NewBackend()).Load(context.Background(), &cfg)
```
- it allows to add optional and non optional files easily

Reply:
Done in 849d41f

Reply:
As I'm still a newbie in Go, is the `New Attribute Object` constructor a common pattern in Go? It sort of feels like an anti-pattern as you'd potentially need to create `x**2` constructors where `x` is the number of optional attributes.

Also `NewOptionalBackend` file sounds like the backend itself is optional (which is kind of is) but doesn't convey the intent that the file can be missing. What's your opinion about that?
To me, the following two solutions seems to be more appropriate:

```go
func (b *Backend) IgnoreMissingFile() *Backend {}
confita.NewLoader(file.NewBackend("").IgnoreMissingfile())
// ---
func NewBackend(path string, opts...Option) *Backend {}
confita.NewLoader(file.NewBackend("", file.IgnoreMissingfile()))
```

Again, I'm still new at Go so please correct me if I don't make sense here 😅